core/sql/optimizer/ItemExpr.cpp

```diff
@@ -14906,3 +14906,80 @@ NABoolean LOBoper::isCovered
   return FALSE;
 }
 
+// Compute the exprssion at compile time. Assume all operands are constants.
+// Return NULL if the computation fails and CmpCommon::diags() may be side-affected.
+ConstValue* ItemExpr::compute(CollHeap* heap)
```

Comment:
I'm slightly confused about what this function is for. The comments and the name ("compute") suggest that it is for evaluation of arbitrary constant expressions, but the comments and variable names inside the function make reference to decoding and keys. But I'm guessing there is  nothing in this function that is decoding-specific or key-specific. Is that correct?